api/src/main/java/uk/gov/hmcts/payment/api/controllers/PaymentGroupController.java

```diff
@@ -332,6 +361,180 @@ private Payment getPayment(PaymentFeeLink paymentFeeLink, String paymentReferenc
+            .bankedDate(DateTime.parse(bulkScanPaymentRequestStrategic.getBankedDate()).withZone(DateTimeZone.UTC).toDate())
+            .build();
+
+        PaymentFeeLink paymentFeeLink = paymentGroupService.addNewBulkScanPayment(payment, paymentGroupReference);
+
+        Payment newPayment = getPayment(paymentFeeLink, payment.getReference());
+
+        allocateThePaymentAndMarkBulkScanPaymentAsProcessed(bulkScanPaymentRequestStrategic, paymentGroupReference, newPayment, headers);
+        return new ResponseEntity<>(paymentDtoMapper.toBulkScanPaymentStrategicDto(newPayment, paymentGroupReference), HttpStatus.CREATED);
+    }
+
+    public void allocateThePaymentAndMarkBulkScanPaymentAsProcessed(BulkScanPaymentRequestStrategic bulkScanPaymentRequestStrategic, String paymentGroupReference, Payment newPayment,
+                                                                    MultiValueMap<String, String> headers) {
+        //Payment Allocation endpoint call
+        PaymentAllocationDto paymentAllocationDto = bulkScanPaymentRequestStrategic.getPaymentAllocationDTO();
+        PaymentAllocation paymentAllocation = PaymentAllocation.paymentAllocationWith()
+            .paymentReference(newPayment.getReference())
+            .paymentGroupReference(paymentGroupReference)
+            .paymentAllocationStatus(paymentAllocationDto.getPaymentAllocationStatus())
+            .reason(paymentAllocationDto.getReason())
+            .explanation(paymentAllocationDto.getExplanation())
+            .userName(paymentAllocationDto.getUserName())
+            .receivingOffice(paymentAllocationDto.getReceivingOffice())
+            .unidentifiedReason(paymentAllocationDto.getUnidentifiedReason())
+            .build();
+        List<PaymentAllocation> paymentAllocationList = new ArrayList<>();
+        paymentAllocationList.add(paymentAllocation);
+        newPayment.setPaymentAllocation(paymentAllocationList);
+
+        //Mark bulk scan payment as processed
+        try {
+            markBulkScanPaymentProcessed(headers, newPayment.getDocumentControlNumber(), "PROCESSED"); // default status PROCESSED
+        } catch (HttpClientErrorException httpClientErrorException) {
+            throw new PaymentException("Bulk scan payment can't be marked as processed for DCN " + newPayment.getDocumentControlNumber() +
+                " Due to response status code as  = " + httpClientErrorException.getMessage());
+        } catch (Exception exception) {
+            throw new PaymentException("Error occurred while processing bulk scan payments with DCN " + newPayment.getDocumentControlNumber() +
+                " Exception message  = " + exception.getMessage());
+        }
+    }
+
+    public ResponseEntity<String> markBulkScanPaymentProcessed(MultiValueMap<String, String> headersMap, String dcn , String status) throws RestClientException {
+        HttpHeaders headers = new HttpHeaders(headersMap);
```

Comment:
Can we not set the required http headers instead of passing through the previous requestor?

Reply:
with these changes bulk scan api called via payhub and not paybubble , propagation is related to that purpose

payhub (request headers) -> bulk scan api

Reply:
see previous comment